File: Sample/Sample/AddNamedUserTableViewController.m
/* Copyright 2017 Urban Airship and Contributors */

@import AirshipKit;

#import "AddNamedUserTableViewController.h"

@interface AddNamedUserTableViewController ()

@end

@implementation AddNamedUserTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.addNamedUserTextField.delegate = self;
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];

    if ([UAirship namedUser].identifier) {
        self.addNamedUserTextField.text = [UAirship namedUser].identifier;
    }
}

- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    return NO;
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField {

    [self.view endEditing:YES];

    NSString *newNamedUser = self.addNamedUserTextField.text;

    // Trim leading whitespace
    NSRange range = [newNamedUser rangeOfString:@"^\\s*" options:NSRegularExpressionSearch];
    NSString *result = [newNamedUser stringByReplacingCharactersInRange:range withString:@""];

    if (result.length) {
        [UAirship namedUser].identifier = result;
    } else {
        textField.text = nil;
        [UAirship namedUser].identifier = nil;
    }

    [[UAirship push] updateRegistration];

    UINavigationController *navigationController = (UINavigationController *) self.parentViewController;
    [navigationController popViewControllerAnimated:YES];

    return YES;
}

@end

File: AirshipDebugKit/AirshipDebugKit/Device-Info/AddNamedUserTableViewController.swift
/* Copyright 2010-2019 Urban Airship and Contributors */

import UIKit
import AirshipKit

class AddNamedUserTableViewController: UITableViewController, UITextFieldDelegate {

    @IBOutlet var addNamedUserTextField: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        self.addNamedUserTextField.delegate = self
    }

    override func viewWillAppear(_ animated: Bool) {
        if ((UAirship.namedUser().identifier) != nil) {
            addNamedUserTextField.text = UAirship.namedUser().identifier
        }
    }

    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        return false
    }

    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        if (textField.text != nil) && (textField.text!.count > 0) {
            UAirship.namedUser().identifier = textField.text
        } else {
            UAirship.namedUser().identifier = nil
        }

        self.view.endEditing(true)
        
        UAirship.push().updateRegistration()

        return navigationController?.popViewController(animated: true) != nil
    }
}
